pkg/apiserver/rest/dualwriter.go

```diff
@@ -30,6 +30,7 @@ type Storage interface {
 	rest.CreaterUpdater
 	rest.GracefulDeleter
 	rest.CollectionDeleter
+	Compare(storageObj, legacyObj runtime.Object) bool
```

Comment:
can you add a comment and explain what it is used for?

IIUC, this is what will be used to verify if the two objects are the same -- and one should be updated